### PR TITLE
KAFKA-7236: Add --under-min-isr option to describe topics command (KIP-351)

### PR DESCRIPTION
**Pull request by KevinLiLu**
[KIP-351](https://cwiki.apache.org/confluence/display/KAFKA/KIP-351%3A+Add+--under-min-isr+option+to+describe+topics+command)

- Add `--under-min-isr` option to `TopicCommand` to report partitions under the configured `min.insync.replicas` value

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/main/scala/kafka/admin/TopicCommand.scala
Comment (vahidhashemian) [inline core/src/main/scala/kafka/admin/TopicCommand.scala]: Since this is just a count, I suggest using `minIsrCount` to avoid confusion with `isr` which is also a list of `Int`.

File: core/src/main/scala/kafka/admin/TopicCommand.scala
Comment (vahidhashemian) [inline core/src/main/scala/kafka/admin/TopicCommand.scala]: Similarly here: `configuredMinIsrCount`?

**Comment by KevinLiLu** [comment]
@viktorsomogyi @gwenshap can you give this a review? Thanks~

File: core/src/test/scala/unit/kafka/admin/TopicCommandWithAdminClientTest.scala
Comment (vahidhashemian) [inline core/src/test/scala/unit/kafka/admin/TopicCommandWithAdminClientTest.scala:547]: Could you also add at least one more unit tests, that verifies the output is empty if there is no under replicated partitions? Alternatively, you can add a unit test that verifies only valid partitions are listed when there is a mix of under-replicated and fully-replicated partitions.

File: core/src/main/scala/kafka/admin/TopicCommand.scala
Comment (vahidhashemian) [inline core/src/main/scala/kafka/admin/TopicCommand.scala]: In the KIP the config is proposed as `under-min-isr-partitions`. Is there a reason you used a different name? I think the name in the KIP (with a dash between `min` and `isr`) looks and reads better.

**Review by vahidhashemian — COMMENTED** [review]
Thank you for the KIP and the PR. Looks good to me overall. I left a few minor comments inline.

File: core/src/main/scala/kafka/admin/TopicCommand.scala
Comment (KevinLiLu) [reply core/src/main/scala/kafka/admin/TopicCommand.scala]: Good catch, thanks!

File: core/src/test/scala/unit/kafka/admin/TopicCommandWithAdminClientTest.scala
Comment (KevinLiLu) [reply core/src/test/scala/unit/kafka/admin/TopicCommandWithAdminClientTest.scala:547]: I've added an additional `testDescribeUnderMinIsrPartitionsMixed`.

**Comment by KevinLiLu** [comment]
Hi @vahidhashemian, thanks for the review! I have added a commit addressing your comments.

I understand that the feature freeze for `2.2.0` was January 31st. Is this KIP bumped to the next release then, or can we still include this minor feature into `2.2.0`?

File: core/src/test/scala/unit/kafka/admin/TopicCommandWithAdminClientTest.scala
Comment (vahidhashemian) [inline core/src/test/scala/unit/kafka/admin/TopicCommandWithAdminClientTest.scala]: nit: Unintentional `println` here?

File: core/src/test/scala/unit/kafka/admin/TopicCommandWithAdminClientTest.scala
Comment (vahidhashemian) [inline core/src/test/scala/unit/kafka/admin/TopicCommandWithAdminClientTest.scala:574]: nit: Please change references to three/3 to four/4 since there are 4 topics now.

**Review by vahidhashemian — COMMENTED** [review]
Thanks for updating the PR. I left a couple of minor nits that you can hopefully fix quickly. Then, I'll approve the PR, and merge once the trunk version is bumped.

Unfortunately, since the cutoff for 2.2 is passed this will be merged to trunk for the following release.

**Review by vahidhashemian — APPROVED** [review]
Thanks for PR. LGTM.

**Review by viktorsomogyi — APPROVED** [review]
Reviewed the PR, LGTM.

**Comment by viktorsomogyi** [comment]
retest this please

**Comment by KevinLiLu** [comment]
retest please

**Comment by KevinLiLu** [comment]
retest this please

**Review by mimaison — APPROVED** [review]
LGTM

**Comment by KevinLiLu** [comment]
Hi @vahidhashemian, thanks for the merge. I see that the trunk version was bumped up after this PR was merged. Did we squeeze this into `2.2.0` release, or is this still pushed back to `2.3.0`?

**Comment by vahidhashemian** [comment]
@KevinLiLu I believe this KIP was originally included in 2.2.0 release plan. So this PR should also be included since the feature freeze is this Friday. @mjsax could you please confirm? Thanks!

**Comment by mjsax** [comment]
KIPs must be merged before feature freeze deadline (cf. https://cwiki.apache.org/confluence/display/KAFKA/Time+Based+Release+Plan) that was two week ago. Thus, this KIP slips for 2.2. Sorry. 

It's already marked as postponed to 2.3 in https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=100827512

**Comment by KevinLiLu** [comment]
@mjsax  no worries, thanks for the clarification!

**Comment by vahidhashemian** [comment]
Thanks for clarifying @mjsax.

**Pull request merged**
